kmath-stat/src/commonMain/kotlin/space/kscience/kmath/samplers/BoxMullerSampler.kt

```diff
@@ -23,8 +24,8 @@ public object BoxMullerSampler : NormalizedGaussianSampler {
         var state = Double.NaN
 
         override fun nextBufferBlocking(size: Int): DoubleBuffer {
-            val xs = generator.nextDoubleBuffer(size)
-            val ys = generator.nextDoubleBuffer(size)
+            val xs = nextDoubleBuffer(size)
```

Comment:
This is a mistake. You are using the default generator instead of provided one. It will give wrong results.